frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1570,6 +1577,80 @@ impl<T: Config> Pallet<T> {
 	}
 }
 
+#[cfg(any(feature = "try-runtime", test))]
+impl<T: Config> Pallet<T> {
+	fn do_try_state() -> Result<(), &'static str> {
+		Self::try_state_snapshot()?;
+		Self::try_state_signed_submissions_map()?;
+		Self::try_state_phase_off()
+	}
+
+	// [`Snapshot`] state check. Invariants:
+	// - [`DesiredTargets`] exist IFF [`Snapshot`] is present.
+	// - [`SnapshotMetadata`] exist IFF [`Snapshot`] is present.
+	fn try_state_snapshot() -> Result<(), &'static str> {
+		let set = <DesiredTargets<T>>::get().is_some() && <SnapshotMetadata<T>>::get().is_some();
+
+		match <Snapshot<T>>::take() {
+            Some(_) if !set => Err("If the snapshot exists, the desired targets and snapshot metadata must also exist."),
+            None if set => Err("If the snapshot does not exists, the desired targets and snapshot metadata should also not exists"),
+            _ => Ok(()),
+		}
+	}
+
+	// [`SignedSubmissionsMap`] state check. Invariants:
+	// - All [`SignedSubmissionIndices`] are present in [`SignedSubmissionsMap`], and no more;
+	// - [`SignedSubmissionNextIndex`] is not present in [`SignedSubmissionsMap`];
+	// - [`SignedSubmissionIndices`] is sorted by election score.
+	fn try_state_signed_submissions_map() -> Result<(), &'static str> {
+		let mut best_score: ElectionScore = Default::default();
+		let indices = <SignedSubmissionIndices<T>>::get();
+
+		for (i, indice) in indices.iter().enumerate() {
+			let submission = <SignedSubmissionsMap<T>>::get(indice.2);
+			if submission.is_none() {
+				return Err("All signed submissions indices must be part of the submissions map")
+			}
+
+			if i == 0 {
+				best_score = indice.0
+			} else {
+				if best_score.strict_threshold_better(indice.0, Perbill::zero()) {
```

Comment:
This would compare every submission with the best score. Would it be better to compare every submission to the last iterated score so we can verify that the whole list is sorted (against current check which only verifies the best submission is at the top)?

Reply:
Good point, fixed!